nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DebugFlow.java

```diff
@@ -0,0 +1,499 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.standard;
+
+import org.apache.http.annotation.ThreadSafe;
+import org.apache.nifi.annotation.behavior.EventDriven;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.flowfile.attributes.CoreAttributes;
+import org.apache.nifi.logging.ComponentLog;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+@ThreadSafe()
+@EventDriven()
+@Tags({"test", "debug", "processor", "utility", "flow", "FlowFile"})
+@CapabilityDescription("The DebugFlow processor aids testing and debugging the FlowFile framework by allowing various "
+        + "responses to be explicitly triggered in response to the receipt of a FlowFile or a timer event without a "
+        + "FlowFile if using timer or cron based scheduling.  It can force responses needed to exercise or test "
+        + "various failure modes that can occur when a processor runs.\n"
+        + "\n"
+        + "When triggered, the processor loops through the appropriate response list (based on whether or not it "
+        + "received a FlowFile).  A response is produced the configured number of times for each pass through its"
+        + "response list, as long as the processor is running.\n"
+        + "\n"
+        + "Triggered by a FlowFile, the processor can produce the following responses."
+        + "  1. transfer FlowFile to success relationship.\n"
+        + "  2. transfer FlowFile to failure relationship.\n"
+        + "  3. rollback the FlowFile without penalty.\n"
+        + "  4. rollback the FlowFile and yield the context.\n"
+        + "  5. rollback the FlowFile with penalty.\n"
+        + "  6. throw an exception.\n"
+        + "\n"
+        + "Triggered without a FlowFile, the processor can produce the following responses."
+        + "  1. do nothing and return.\n"
+        + "  2. throw an exception.\n"
+        + "  3. yield the context.\n")
```

Comment:
I wonder if there's an abbreviated version of the doc for capability description, and move this out to an additional details page